modules/LinearBoltzmannSolver/IterativeOperations/lbs_compute_fission_production.cc

```diff
@@ -30,9 +30,8 @@ double SteadySolver::ComputeFissionProduction(const std::vector<double>& phi)
       const double IntV_ShapeI = cell_matrices.Vi_vectors[i];
 
       //=============================== Loop over groups
-      //TODO: This should be nu
       for (size_t g = first_grp; g <= last_grp; ++g)
-        local_production += xs.nu_prompt_sigma_f[g] *
+        local_production += xs.nu_sigma_f[g] *
                             phi[uk_map + g] *
                             IntV_ShapeI;
     }//for node
```

Comment:
Should this not be something like this:
```c++
      if (not options.use_precursors)
      {
        for (size_t g = first_grp; g <= last_grp; ++g)
          local_production += xs.nu_sigma_f[g] *
                              phi[uk_map + g] *
                              IntV_ShapeI;
      }
      else
      {
        for (size_t g = first_grp; g <= last_grp; ++g)
          local_production += xs.nu_prompt_sigma_f[g] *
                              phi[uk_map + g] *
                              IntV_ShapeI;
        for (const auto& precursor : xs.precursors)
          for (size_t g = first_grp; g <= last_grp; ++g)
            local_production += precursor.fractional_yield *
                                xs.nu_delayed_sigma_f[g] *
                                phi[uk_map + g] *
                                IntV_ShapeI;
      }
```

Reply:
Hmmm. I need to think about this. Normally, these are equivalent because $\nu = \nu^p + \nu^d$ and $\sum_j \gamma_j = 1$. However, this may need to be changed.

Reply:
Changed to 

```c++
for (size_t g = first_grp; g <= last_grp; ++g)
      {
        const auto& prod_g = xs.production_matrix[g];
        for (size_t gp = 0; gp <= last_grp; ++gp)
          local_production += prod_g[gp] *
                              phi[uk_map + gp] *
                              IntV_ShapeI;

        if (options.use_precursors)
          for (unsigned int j = 0; j < xs.num_precursors; ++j)
            local_production += xs.nu_delayed_sigma_f[g] *
                                phi[uk_map + g] *
                                IntV_ShapeI;
      }
```

Using the production matrix we don't need addition checks because it is always either total or prompt, depending on what was specified. If total was specified, `options.use_precursors` will always be false.

Reply:
BUG! @zachhardy The two routines in this file need the following check:
```c++
const bool fission_avail = xs.is_fissionable
```
and use just like we do in the `SetSource` routines, otherwise they crash if a material is encountered without fission. I found this bug by running `tests/BigTests/c5g7/X_Test1.lua`.

Reply:
I was able to fix this from my end.